### ISSUE TITLE
About different sequence input

### ISSUE THREAD

**Issue by LemoJa** [locuslab/TCN#28]
I have a totally different sequence, the smallest length is about 100 words, the max is about 5000. I attempt to padding zero to the same length of 5000, but the classified result is terrible. but if I just input different size, that's means keep the original and make the batch_size just 1, that's works well. I don't know why this happens.

**Comment by jerrybai1995** [comment]
If you want to train by batch, I suggest grouping inputs of similar lengths together, and then do the padding (for instance, inputs of 90-100 should be in the same batch). From my experience, when the padding is too long compared to the original sequence (e.g., 4900 zeros and then 100 non-zeros) it's harder for the dilated convolution to learn without good initialization of the weights.

**Comment by LemoJa** [comment]
thanks for your reply, I will try it.

**Comment by deancorleone00** [comment]
How to implement the padding that is varies by batch?

**Comment by jerrybai1995** [comment]
@deancorleone00 Paddings that vary by batch shouldn't be an issue as convolution operations are supposed to be able to be applied on any input size (e.g., a 2D convolution can be applied on images of all sizes).

**Comment by deancorleone00** [comment]
Hi, I have done grouping samples with similar length and pad them for training. But what about evaluation for losses? Can I pad all my samples and run `model(X_test)` to evaluate all at once?

**Comment by jerrybai1995** [comment]
Sure, you can do it in the same way as in training; but you don't need to if you pass in one sample at a time. Padding only ensures that all samples in a batch are extended to equal lengths.

**Comment by johnsyin** [comment]
How to pass in one sample at a time? As per epoch I want to get cross entropy loss of all my evaluation data that has different length. Sorry if it sounds dumb, I am new to pytorch.

**Comment by jerrybai1995** [comment]
Cross entropy is defined on one train/test sample. You can pass in multiple samples at a time (which is called "a minibatch"), but you can also just pass in one. Just set batch size = 1.

**Comment by johnsyin** [comment]
Thanks! I will try!

**Comment by ChunyangCLi** [comment]
Usually we shuffle data and randomly put them into different batches, so that batches can be representations of the dataset. Batches wouldn't be random if they are based on different sequence lengths. What do you think on this problem?
